docs/books/learning_bash/01-first-script.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Bash - First script
+author: Antoine Le Morvan
+contributors: Steven Spencer
+update: 29-mar-2022
+---
+
+# Bash - First script
+
+In this chapter you will learn how to write your first script in bash.
+
+****
+
+**Objectives**: In this chapter you will learn how to:
+
+:heavy_check_mark: Write your first script in bash;
+:heavy_check_mark: Execute your first script;
+:heavy_check_mark: Specify which shell to use with the so-called shebang;
+
+:checkered_flag: **linux**, **script**, **bash**
+
+**Knowledge**: :star:     
+**Complexity**: :star:
+
+**Reading time**: 10 minutes
+
+****
+
+## My first script
+
+To start writing a shell script, it is convenient to use a text editor that supports syntax highlighting.
+
+`vim`, for example, is a good tool for this.
+
+The name of the script should respect some rules:
+
+* no names of existing commands ;
+* only alphanumeric characters, i.e. no accented characters or spaces;
+* extension .sh to indicate that it is a shell script.
+
+```
+#!/usr/bin/env bash
```

Comment:
Yes me too. I have always put `#!/bin/bash`.
In the case of python for example, 
`#!/usr/bin/env python` will use the python version defined by alternative.
I have read that it's good to do this, because it does not matter of the path to bash (in your case #!/usr/bin/bash)

Reply:
Using env in the shebang line has been around a long time, and was designed to allow scripts to work portably on *nix's that had executables in alternate dirs (like /usr/local/bin, /opt/bin, etc). Like many "improvements" regarding abstraction it didn't really catch on and many devs/admins never even knew about it, another classic example is using "info" rather than "man", or surrounding program calls in scripts with parenthesis (for security). Later, others who either were unaware of, or just dissagreed with this method developed other solutions like `alternatives`, "environment-modules", and "direnv", while some devs devised specific program solutions like python's "virtualenv"  and "anaconda" etc. `env` has been the std for a very long time, but was primarily used to temporarily alter environment settings to pass to autotools utils like `configure`, as well as `gcc`. The bottom line is that what you use for shebang depends on what the system admins implement systemwide, and/or per user and sadly there is currently no real "best-practice" in any universal sense because there is so much variance in system setup techniques, but using `env` has gained some traction and offers the most flexibility for admins and users. I have used it in the shebang for 15 or so years and not had any major issues, however I was the admin for all the systems I used and set things up so that there would be consistency among the various elements (alternatives, environment-modules, $PATH, /etc/skel/* , etc)

Reply:
Hey thanks @NezSez for this explanation!
The review is really a good thing 👍